Solutions/Marain.Tenancy.Cli/Marain/Tenancy/Cli/Commands/List.cs

```diff
@@ -0,0 +1,125 @@
+﻿// <copyright file="List.cs" company="Endjin Limited">
+// Copyright (c) Endjin Limited. All rights reserved.
+// </copyright>
+
+namespace Marain.Tenancy.Cli.Commands
+{
+    using System;
+    using System.Collections.Generic;
+    using System.IO;
+    using System.Linq;
+    using System.Threading.Tasks;
+    using ConsoleTables;
+    using Corvus.Tenancy;
+    using McMaster.Extensions.CommandLineUtils;
+
+    /// <summary>
+    /// Lists children of the specified tenant.
+    /// </summary>
+    [Command(Name = "list", Description = "List tenants.")]
+    public class List
+    {
+        private readonly ITenantProvider tenantProvider;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="List"/> class.
+        /// </summary>
+        /// <param name="tenantProvider">The tenant provider that will be used to retrieve the information.</param>
+        public List(ITenantProvider tenantProvider)
+        {
+            this.tenantProvider = tenantProvider;
+        }
+
+        /// <summary>
+        /// Gets or sets the tenant whose children should be retrieved.
+        /// </summary>
+        [Option(CommandOptionType.SingleOrNoValue, ShortName = "t", LongName = "tenant", Description = "The Id of the tenant to retrieve children for. Leave blank to retrieve children of the root tenant.")]
+        public string TenantId { get; set; }
+
+        /// <summary>
+        /// Gets or sets a value indicating whether or not details should be loaded for the tenants.
+        /// </summary>
+        [Option(CommandOptionType.MultipleValue, ShortName = "p", LongName = "property", Description = "The names of properties to include.")]
+        public string[] IncludeProperties { get; set; }
+
+        /// <summary>
+        /// Executes the command.
+        /// </summary>
+        /// <param name="app">The current <c>CommandLineApplication</c>.</param>
+        /// <returns>A <see cref="Task"/> representing the asynchronous operation.</returns>
+        public async Task<int> OnExecute(CommandLineApplication app)
+        {
+            if (string.IsNullOrEmpty(this.TenantId))
+            {
+                this.TenantId = this.tenantProvider.Root.Id;
+            }
+
+            string continuationToken = null;
+
+            var childTenantIds = new List<string>();
+
+            do
+            {
+                TenantCollectionResult children = await this.tenantProvider.GetChildrenAsync(this.TenantId, 2, continuationToken).ConfigureAwait(false);
```

Comment:
A batch size of 2 seems like an odd choice. Where did that come from?

(Also, this is a scenario where I'd consider using a named argument, because I had to open up `Corvus.Tenancy` in Visual Studio to find out what the argument actually does.)

Reply:
This came from "testing to make sure that using the continuation token works when there's only a small number of tenants".

I have changed back to a more appropriate batch size.